internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -344,9 +343,24 @@ func resourceKustoClusterCreateUpdate(d *pluginsdk.ResourceData, meta interface{
 		TrustedExternalTenants: expandTrustedExternalTenants(d.Get("trusted_external_tenants").([]interface{})),
 	}
 
-	if v, ok := d.GetOk("virtual_network_configuration"); ok {
+	oldLength, newLength := d.GetChange("virtual_network_configuration.#")
+
+	if newLength.(int) > 0 {
+		v := d.Get("virtual_network_configuration")
 		vnet := expandKustoClusterVNET(v.([]interface{}))
 		clusterProperties.VirtualNetworkConfiguration = vnet
+		clusterProperties.VirtualNetworkConfiguration.State = pointer.To(clusters.VnetStateEnabled)
+	} else if oldLength.(int) > 0 {
+		oldSubnetId, _ := d.GetChange("virtual_network_configuration.0.subnet_id")
+		oldEnginePublicIpId, _ := d.GetChange("virtual_network_configuration.0.engine_public_ip_id")
+		oldDataManagementPublicIpId, _ := d.GetChange("virtual_network_configuration.0.data_management_public_ip_id")
+
+		clusterProperties.VirtualNetworkConfiguration = &clusters.VirtualNetworkConfiguration{
+			SubnetId:                 oldSubnetId.(string),
+			EnginePublicIPId:         oldEnginePublicIpId.(string),
+			DataManagementPublicIPId: oldDataManagementPublicIpId.(string),
+			State:                    pointer.To(clusters.VnetStateDisabled),
+		}
```

Comment:
I think we should be able to use the `existing` resources model to pull this value directly from Azure instead of pulling it from the state file? WDYT?

```suggestion
	// If the value in the config is nil we should, pull the
	// network configuration from the existing resource...
	if v, ok := d.GetOk("virtual_network_configuration"); ok {
		if vnetConfig := expandKustoClusterVNET(v.([]interface{})); vnetConfig != nil {
			clusterProperties.VirtualNetworkConfiguration = vnetConfig
			clusterProperties.VirtualNetworkConfiguration.State = pointer.To(clusters.VnetStateEnabled)
		}
	} else {
		if model := existing.Model; model != nil && model.Properties.VirtualNetworkConfiguration != nil {
			clusterProperties.VirtualNetworkConfiguration = model.Properties.VirtualNetworkConfiguration
			clusterProperties.VirtualNetworkConfiguration.State = pointer.To(clusters.VnetStateDisabled)
			clusterProperties.PublicNetworkAccess = pointer.To(clusters.PublicNetworkAccessDisabled)
		}
	}
```